### PR TITLE
Feature/approle

### PR DESCRIPTION
**Pull request by TerryHowe**

### CODE REVIEW

**Comment by TerryHowe** [comment]
@drewmullen if you get a chance, trying to do things a bit more legit here

File: ansible/modules/hashivault/hashivault_approle_role.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+            current_state = client.get_role(name, mount_point=mount_point)
+            changed = False
+            missing = []
+            for key in desired_state:
+                if key in current_state:
+                    if current_state[key] != desired_state[key]:
+                        changed = True
+                else:
+                    missing.append(key)
+            if changed:
```
Comment (drewmullen) [inline ansible/modules/hashivault/hashivault_approle_role.py]: you also want to catch for check_mode before actually executing the api call. when check_mode is enabled you dont want to actually execute the api call, just return "changed". ex: `if changed and state == 'present' and not module.check_mode:`

File: ansible/modules/hashivault/hashivault_approle_role.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    argspec['bound_cidr_list'] = dict(required=False, type='list')
+    argspec['policies'] = dict(required=False, type='list', default=[])
+    argspec['secret_id_num_uses'] = dict(required=False, type='str')
+    argspec['secret_id_ttl'] = dict(required=False, type='str')
+    argspec['token_num_uses'] = dict(required=False, type='int')
+    argspec['token_ttl'] = dict(required=False, type='str')
+    argspec['token_max_ttl'] = dict(required=False, type='str')
+    argspec['period'] = dict(required=False, type='str')
+    argspec['enable_local_secret_ids'] = dict(required=False, type='bool')
+    module = hashivault_init(argspec)
```
Comment (drewmullen) [inline ansible/modules/hashivault/hashivault_approle_role.py]: if you support check mode (see below) you can pass it to the `_init()`. ex: `hashivault_init(argspec, supports_check_mode)`

**Comment by drewmullen** [comment]
love it! a few comments about supporting check mode but it looks great otherwise. good work!

**Comment by TerryHowe** [comment]
I'll add check_mode support on another PR to keep things simple. This PR is too big as it is.

**Pull request merged**